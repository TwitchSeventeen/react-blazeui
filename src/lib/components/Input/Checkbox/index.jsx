import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Checkbox(props) {
  const {
    className, success, error, radio, dataName, ...other
  } = props;
  const classes = classNames(
    { 'c-field--success': success },
    { 'c-field--error': error },
    className,
  );

  if (radio) {
    return (
      <input
        {...other}
        className={classes}
        type="radio"
        data-name={dataName}
      />
    );
  }

  return (
    <input
      {...other}
      className={classes}
      type="checkbox"
      data-name={dataName}
    />
  );
}

Checkbox.propTypes = {
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines successful styling
   */
  success: PropTypes.bool,
  /**
   * Defines error styling
   */
  error: PropTypes.bool,
  /**
   * Defines a radio input
   */
  radio: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Checkbox.defaultProps = {
  className: null,
  success: false,
  error: false,
  radio: false,
  dataName: 'input-checkbox',
};

export default Checkbox;
