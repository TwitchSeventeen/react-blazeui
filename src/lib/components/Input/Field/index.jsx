import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Field(props) {
  const {
    className, iconLeft, iconRight, children, fixed, dataName, ...other
  } = props;
  const classes = classNames(
    'o-field',
    { 'o-field--icon-left': iconLeft },
    { 'o-field--icon-right': iconRight },
    { 'o-field--fixed': fixed },
    className,
  );

  return (
    <div
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </div>
  );
}

Field.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines a left icon
   */
  iconLeft: PropTypes.bool,
  /**
   * Defines a right icon
   */
  iconRight: PropTypes.bool,
  /**
   * Defines a fixed with field
   */
  fixed: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Field.defaultProps = {
  className: null,
  iconLeft: false,
  iconRight: false,
  children: null,
  fixed: false,
  dataName: 'input-field',
};

export default Field;
