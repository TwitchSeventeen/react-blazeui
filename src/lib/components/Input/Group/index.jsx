import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Group(props) {
  const {
    className, children, stacked, as: Component, rounded, roundedLeft, roundedRight, dataName, ...other
  } = props;
  const classes = classNames(
    'c-input-group',
    { 'c-input-group--stacked': stacked },
    { 'c-input-group--rounded': rounded },
    { 'c-input-group--rounded-left': roundedLeft },
    { 'c-input-group--rounded-right': roundedRight },
    className,
  );

  return (
    <Component
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </Component>
  );
}

Group.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines a stacked group of Inputs
   */
  stacked: PropTypes.bool,
  /**
   * Defines a group with rounded corners
   */
  rounded: PropTypes.bool,
  /**
   * Defines a group with rounded left corners
   */
  roundedLeft: PropTypes.bool,
  /**
   * Defines a group with rounded right corners
   */
  roundedRight: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
  /**
   * Element type to render as
   */
  as: PropTypes.elementType,
};

Group.defaultProps = {
  className: null,
  children: null,
  stacked: false,
  rounded: false,
  roundedLeft: false,
  roundedRight: false,
  dataName: 'input-group',
  as: 'div',
};

export default Group;
