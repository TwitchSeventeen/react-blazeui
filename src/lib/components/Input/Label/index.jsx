import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Label(props) {
  const {
    className, choice, children, inputId, ...other
  } = props;
  const classes = classNames(
    'c-field',
    { 'c-field--choice': choice },
    { 'c-field': choice },
    className,
  );

  return (
    <label {...other} className={classes} htmlFor={inputId}>
      { children }
    </label>
  );
}

Label.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines a label with a radio or checkbox
   */
  choice: PropTypes.bool,
  /**
   * ID to link to input for accessibility
   */
  inputId: PropTypes.string,
};

Label.defaultProps = {
  children: null,
  className: null,
  choice: false,
  inputId: null,
};

export default Label;
