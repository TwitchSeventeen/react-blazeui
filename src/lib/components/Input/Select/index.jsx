import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Select (props) {
  const { children, className, success, error } = props;
  const classes = classNames(
      'c-field',
      { 'c-field--success': success },
      { 'c-field--error': error },
      className
  );

  return (
      <select {...props} className={classes}>
        { children }
      </select>
  );
}

Select.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines successful styling
   */
  success: PropTypes.bool,
  /**
   * Defines error styling
   */
  error: PropTypes.bool,
};

Select.defaultProps = {
  children: null,
  className: null,
};

export default Select;
