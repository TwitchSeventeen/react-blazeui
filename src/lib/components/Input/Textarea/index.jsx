import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Textarea(props) {
  const {
    children, className, success, error, dataName, ...other
  } = props;
  const classes = classNames(
    'c-field',
    { 'c-field--success': success },
    { 'c-field--error': error },
    className,
  );

  return (
    <textarea
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </textarea>
  );
}

Textarea.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines successful styling
   */
  success: PropTypes.bool,
  /**
   * Defines error styling
   */
  error: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Textarea.defaultProps = {
  children: null,
  className: null,
  success: false,
  error: false,
  dataName: 'input-textarea',
};

export default Textarea;
