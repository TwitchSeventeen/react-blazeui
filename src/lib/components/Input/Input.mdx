---
name: Input
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Input from './index';

# Input

From single input controls like text boxes to full form patterns.

## Text Input
<Playground>
    <Input placeholder="Type in here..." />
    <br />
    <br />
    <Input.Textarea placeholder="Type in here..."></Input.Textarea>
    <br />
    <br />
    <Input.Select>
        <option>Choose me...</option>
    </Input.Select>
</Playground>

## Colours
<Playground>
    <Input placeholder="success" success />
    <br />
    <br />
    <Input placeholder="error" error />
    <br />
    <br />
    <Input placeholder="disabled" disabled />
</Playground>

## Properties
<PropsTable of={Input} />