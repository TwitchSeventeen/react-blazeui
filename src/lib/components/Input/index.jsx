import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Textarea from './Textarea';
import Select from './Select';
import Field from './Field';
import Group from './Group';
import Checkbox from './Checkbox';
import Label from './Label';

function Input(props) {
  const {
    className, success, error, size, dataName, ...other
  } = props;
  const classes = classNames(
    'c-field',
    { 'c-field--success': success },
    { 'c-field--error': error },
    { [`u-${size}`]: size },
    className,
  );

  return (
    <input
      {...other}
      className={classes}
      data-name={dataName}
    />
  );
}

Input.propTypes = {
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * The type of input
   */
  type: PropTypes.string,
  /**
   * Defines successful styling
   */
  success: PropTypes.bool,
  /**
   * Defines error styling
   */
  error: PropTypes.bool,
  /**
   * Size of the input
   */
  size: PropTypes.oneOf([
    'xsmall',
    'small',
    'medium',
    'large',
    'xlarge',
    'super',
  ]),
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Input.defaultProps = {
  className: null,
  success: false,
  error: false,
  type: 'text',
  size: null,
  dataName: 'input',
};

Input.Textarea = Textarea;
Input.Select = Select;
Input.Field = Field;
Input.Group = Group;
Input.Checkbox = Checkbox;
Input.Label = Label;

export default Input;
