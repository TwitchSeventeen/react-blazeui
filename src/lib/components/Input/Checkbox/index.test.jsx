import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Input from '../index';

describe('Checkbox', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Input.Checkbox).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Input.Label choice>
        <Input.Checkbox />
        {' '}
        Tick the box
      </Input.Label>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
