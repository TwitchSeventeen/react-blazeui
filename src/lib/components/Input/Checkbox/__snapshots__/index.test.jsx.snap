// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<DocumentFragment>
  <label
    class="c-field c-field--choice c-field"
    data-name="input-label"
  >
    <input
      class=""
      data-name="input-checkbox"
      type="checkbox"
    />
     Tick the box
  </label>
</DocumentFragment>
`;
